src/dune_rules/pkg_rules.ml

```diff
@@ -748,17 +1015,27 @@ module Action_expander = struct
       | Sys_ocaml_version ->
         sys_poll_var (fun { sys_ocaml_version; _ } -> sys_ocaml_version)
       | Build -> Memo.return [ Value.Dir (Path.build paths.source_dir) ]
-      | Prefix -> Memo.return [ Value.Dir (Path.build paths.target_dir) ]
+      | Prefix ->
+        Memo.return
+          [ (match override_pform.prefix with
+             | None -> Value.Dir (Path.build paths.target_dir)
```

Comment:
Why can't you just set `paths.target_dir` for this package instead?

Reply:
Because `paths.target_dir` is a `Path.Build.t` but I need to override it with a path outside the build directory. I guess we could change `Paths.t` to use `Path.t` rather than `Path.Build.t`, but then we would lose the guarantee that all the paths are (usually) within the build directory. Do you think it would be worth it to make this change to `Paths.t`?

Reply:
I had a go at changing the types in `Paths.t` to be `Path.t` but ran into an issue where a `Paths.t` is used to create actions, and a lot of the actions constructors expect `Path.Build.t`. I don't want to convert from `Path.t` to `Path.Build.t` in these places because then we'd have to think about what to do when the conversion fails which would complicate the logic.

Reply:
Okay, I will try doing something to improve this myself.

> because then we'd have to think about what to do when the conversion fails which would complicate the logic.

I think that's exactly the kind of a problem that's already present in this code. One has to remember if the package is a toolchain to access the package fields correctly, and it's very error prone.